HW0/P5/P5.txt

```diff
@@ -1 +1,23 @@
-# Your answers here.
+# Qing Zhao (HUID: 80985053)’s answer for P5
+
+1. 255 seconds.
+
+2. The time for each of them to count 256 / 8 = 32 bags is 31 s; summing up 8 numbers takes at least 3s (2 ^ 3 = 8). Therefore, 8 cashiers can do this in 34 seconds.
+
+3. 8 seconds.
+
+4. See ‘P5.png’
+
+5. With infinite employees, it will take 8 + 7 = 15 seconds to count 256 bags;
+
+6. 1 bag each: 256 + log2(256) + log2(256) = 272 seconds;
+   2 bags each: 128 + log2(128) + log2(128) = 142 seconds;
+
```

Comment:
If I'm thinking about this problem correctly, the previous employees can be adding and communicating while the next employees are receiving bags so once the final bag is handed out the last employee (with 16 employees) takes 15 more seconds to add (assuming an employee can't receive bags and add at the same time), one second to communicate with the others, and one second to add their total to theirs. So in the end it should take 256+17 = 273seconds.

With 2 bags each, similar logic leads to 128+17 = 145 seconds.